File: Software/NodeJS/examples/EasyMotors.js
const sleep = require('sleep');
const EasyGopigo = require('../lib/easyGopigo3');

const gpg = new EasyGopigo();

/*
console.log('Move the motors forward freely for 1 second.');
gpg.forward();
sleep.sleep(1);

console.log('Stop the motors for 1 second.');
gpg.stop();
sleep.sleep(1);
*/

console.log('Press any key to exit');

process.stdin.setRawMode(true);
process.stdin.resume();
process.stdin.on('data', () => {
    gpg.stop();
    gpg.resetAll();
    process.exit(0);
});

console.log('Drive the motors 50 cm and then stop.');
gpg.driveCm(50, () => {
    console.log('Wait 1 second.');
    sleep.sleep(1);

    console.log('Turn right 1 second.');
    gpg.right();
    sleep.sleep(1);

    console.log('Turn left 1 second.');
    gpg.left();
    sleep.sleep(1);

    console.log('Stop!');
    gpg.stop();

    console.log('Done!');
});

File: Software/Python/Examples/easy_Motors.py
#!/usr/bin/env python
#
# https://www.dexterindustries.com/GoPiGo/
# https://github.com/DexterInd/GoPiGo3
#
# Copyright (c) 2017 Dexter Industries
# Released under the MIT license
# (http://choosealicense.com/licenses/mit/).
#
# For more information see
# https://github.com/DexterInd/GoPiGo3/blob/master/LICENSE.md
#
# This code is an example for controlling the GoPiGo3 motors.  This uses
# the EasyGoPiGo3 library.  You can find more information on the library
# here:  http://gopigo3.readthedocs.io/en/latest/api-basic.html#easygopigo3
#
# Results:  The GoPiGo3 will move forward for 2 seconds, and then
# backward for 2 second.


# import the time library for the sleep function
import time
# import the GoPiGo3 drivers
import easygopigo3 as easy

# Create an instance of the GoPiGo3 class.
# GPG will be the GoPiGo3 object.
gpg = easy.EasyGoPiGo3()

print("Move the motors forward freely for 1 second.")
gpg.forward()
time.sleep(1)

print("Stop the motors for 1 second.")
gpg.stop()
time.sleep(1)

print("Drive the motors 50 cm and then stop.")
gpg.drive_cm(50, True)

print("Wait 1 second.")
time.sleep(1)

print("Turn right 1 second.")
gpg.right()
time.sleep(1)

print("Turn left 1 second.")
gpg.left()
time.sleep(1)

print("Stop!")
gpg.stop()
print("Done!")
